### ISSUE TITLE
profiling

### ISSUE THREAD

**Issue by ksjogo** [ksjogo/emacs-libgit2#4]
Which commands are slow and need improvement?

**Comment by vermiculus** [comment]
Are we talking about Magit commands? If that's the case, @tarsius would be the expert on that. I seem to recall his reports that `reflog`-based functionality is slow.

Selfishly, I'd like to see improvement in any command that calls `git` multiple times -- `git` is very slow on Windows.

**Comment by ksjogo** [comment]
Yeah, I plan to advise/overwrite magit functions for the moment. Tarsius gave advice on some magit thread, I just need to find more time developing stuff.

Could you try building the new cmake setup on windows, I am currently back on GNU/Linux and osx and don't have access.

**Comment by vermiculus** [comment]
Are you interested in building under cygwin or native windows?

**Comment by ksjogo** [comment]
Probably depends on how people usually compile/get binaries for Windows. I think the non-cygwin mingw builds but am not sure about that, would be great, if you could check/fix these.

**Comment by vermiculus** [comment]
One possibility for seeing which things magit actually calls the most: set `magit-git-executable` to a script that logs its usage to a file before passing them to git.

**Comment by ksjogo** [comment]
Good idea.
We also need some kind of demo git repository on which we can run other test cases.

**Comment by tarsius** [comment]
I would recommend looking at `magit-git.el` which contains a lot of low-level git stuff. I would like to use libgit for most things in there.

**Comment by tarsius** [comment]
> I seem to recall his reports that `reflog`-based functionality is slow.

It was `rev-parse`. `config` might also be a good place to start.
